Clojure/tests/syntax_test_clojure.clj

```diff
@@ -0,0 +1,2007 @@
+;     ^ comment.line.clojure punctuation.definition.comment
+
+; ## Unaffected
+
+  'nil (true) (nil)
+; ^ constant.other.symbol.reader-macro.clojure
+;  ^^^ constant.language.clojure
+;      ^ punctuation.section.parens.begin.clojure
+;       ^^^^ constant.language.clojure
+;           ^ punctuation.section.parens.end.clojure
+
+; ## No highlighting
+
+  nill nil- -nil nil?
+; ^^^^^^^^^^^^^^^^^^^- constant
+
+
+
+; # Numbers
+
+  1234 1234N +1234 +1234N -1234 -1234N
+; ^^^^ constant.numeric.clojure
+;     ^- constant
+;      ^^^^^ constant.numeric.clojure
+;            ^^^^^ constant.numeric.clojure
+;                  ^^^^^^ constant.numeric.clojure
+;                         ^^^^^ constant.numeric.clojure
+;                               ^^^^^^ constant.numeric.clojure
+  0x1234af 0x1234afN 0X1234AF 0X1234AFN
+; ^^^^^^^^ constant.numeric.clojure
+;         ^- constant
+;          ^^^^^^^^^ constant.numeric.clojure
+;                    ^^^^^^^^ constant.numeric.clojure
+;                             ^^^^^^^^^ constant.numeric.clojure
+  +0x1234af +0x1234afN +0X1234AF +0X1234AFN
+; ^^^^^^^^^ constant.numeric.clojure
+;          ^- constant
+;           ^^^^^^^^^^ constant.numeric.clojure
+;                      ^^^^^^^^^ constant.numeric.clojure
+;                                ^^^^^^^^^^ constant.numeric.clojure
+  -0x1234af -0x1234afN -0X1234AF -0X1234AFN
+; ^^^^^^^^^ constant.numeric.clojure
+;          ^- constant
+;           ^^^^^^^^^^ constant.numeric.clojure
+;                      ^^^^^^^^^ constant.numeric.clojure
+;                                ^^^^^^^^^^ constant.numeric.clojure
+  2r1010 16r1234af 32r1234az 2R1010 16R1234AF 32R1234AZ
+; ^^^^^^ constant.numeric.clojure
+;       ^- constant
+;        ^^^^^^^^^ constant.numeric.clojure
+;                  ^^^^^^^^^ constant.numeric.clojure
+;                            ^^^^^^ constant.numeric.clojure
+;                                   ^^^^^^^^^ constant.numeric.clojure
+;                                             ^^^^^^^^^ constant.numeric.clojure
+  +2r1010 +16r1234af +32r1234az +2R1010 +16R1234AF +32R1234AZ
+; ^^^^^^^ constant.numeric.clojure
+;        ^- constant
+;         ^^^^^^^^^^ constant.numeric.clojure
+;                    ^^^^^^^^^^ constant.numeric.clojure
+;                               ^^^^^^^ constant.numeric.clojure
+;                                       ^^^^^^^^^^ constant.numeric.clojure
+;                                                  ^^^^^^^^^^ constant.numeric.clojure
+  -2r1010 -16r1234af -32r1234az -2R1010 -16R1234AF -32R1234AZ
+; ^^^^^^^ constant.numeric.clojure
+;        ^- constant
+;         ^^^^^^^^^^ constant.numeric.clojure
+;                    ^^^^^^^^^^ constant.numeric.clojure
+;                               ^^^^^^^ constant.numeric.clojure
+;                                       ^^^^^^^^^^ constant.numeric.clojure
+;                                                  ^^^^^^^^^^ constant.numeric.clojure
+  0/10 10/20 30/0
```

Comment:
What does the `/` represent?

Reply:
In Clojure, `N/M`, where N and M are integers, is a ratio literal. Ratios are a special numeric data type.

Reply:
How about adding a `punctuation.separator.ratio-literal.clojure` or similar scope to the slash?

Reply:
Sure. `punctuation.separator.ratio.clojure` sounds good to me.